### PR TITLE
[BACKPORT] Added tests for SelectorMode and JCacheDetector

### PR DESCRIPTION
**Pull request by vbekiaris**

### CODE REVIEW

**Comment by vbekiaris** [comment]
Updated with `JCacheDetector` NPE fix and test by @Donnerbart

**Comment by vbekiaris** [comment]
Test failure at com.hazelcast.instance.NodeStateTest.multipleShutdowns_Allowed
```
java.lang.IllegalStateException: Node failed to start!
	at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:133)
	at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:160)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:143)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:112)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:99)
	at com.hazelcast.instance.NodeStateTest.multipleShutdowns_Allowed(NodeStateTest.java:81)
```

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
Previous test failure in com.hazelcast.concurrent.atomiclong.AtomicLongAdvancedTest.testAtomicLongSpawnNodeInParallel ( #6055 ?)

**Comment by vbekiaris** [comment]
verify

File: hazelcast/src/test/java/com/hazelcast/nio/tcp/nonblocking/SelectorModeTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class SelectorModeTest {
+
+
+    @Test
+    public void getConfiguredValue_whenSelectWithFix() throws Exception {
+        System.setProperty("hazelcast.io.selectorMode", "selectwithfix");
+        assertEquals(SelectorMode.SELECT_WITH_FIX, SelectorMode.getConfiguredValue());
+        System.setProperty("hazelcast.io.selectorMode", "");
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/nio/tcp/nonblocking/SelectorModeTest.java]: I guess you want to store the original value and restore it after test. in a finally block?

**Comment by vbekiaris** [comment]
verify

**Comment by jerrinot** [comment]
:+1: can be merged on green

**Comment by vbekiaris** [comment]
Thank you @jerrinot

**Pull request merged**